cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -388,4 +393,175 @@ public void setVersion(VersionIdentifier version, boolean hint) {
+   * @param toolVersionToCheck the {@link String} of the tool version that is installed and needs to be checked
+   * in the Json file to find the right version of the dependency to be installed
+   * @return {@link String} the Version of the dependency with which the tool works correctly.
+   */
+  private String findDependencyVersionFromJson(JsonNode toolVersions, String toolVersionToCheck) {
+    String requiredDependencyVersion = null;
+
+    // Iterate through the fields of the Json file
+    Iterator<Map.Entry<String, JsonNode>> fields = toolVersions.fields();
+    int[] targetVersion = parseVersion(toolVersionToCheck);
+
+    OuterLoop:
+    while (fields.hasNext()) {
+
+      Map.Entry<String, JsonNode> entry = fields.next();
+      String versionKey = entry.getKey();
+      int[] foundToolVersion = parseVersion(versionKey); // Found version when iterating the Json file
+
+      if (isEqualOrLess(foundToolVersion, targetVersion)) {
+        JsonNode dependencies = entry.getValue();
+        for (JsonNode dependencyNode : dependencies) {
+          if (getDependencyName().equals(dependencyNode.get("dependency").asText())) {
+            requiredDependencyVersion = dependencyNode.get("MinVersion").asText();
+            // TODO: Add logic to handle MaxVersion if needed
+            break OuterLoop; // Stop the loop when a matching Java version is found
+          }
+        }
+      }
+    }
+
+    return requiredDependencyVersion;
+  }
+
+  /**
+   * Method to parse the versions because they normally contain dots, like 10.1.14 should be separated into
+   * 10, 1 and 14
+   *
+   * @param versionString the {@link String} of the Version in its normal form
+   * @return Array of integers with the integers contained in the Version String.
+   */
+  private int[] parseVersion(String versionString) {
+
+    String[] versionComponents = versionString.split("\\.");
+
+    int[] version = new int[versionComponents.length];
+    for (int j = 0; j < versionComponents.length; j++) {
+      version[j] = Integer.parseInt(versionComponents[j]);
+    }
+    return version;
+  }
+
+  /**
+   * Method to compare two versions after they are parsed, so basically each Integer part of a version is compared with
+   * the corresponding part of the other Version.
+   * 
+   * @param currentToolVersion the Array of integers of the current tool version that is parsed
+   * @param targetVersion the Array of integers of the target version, that is also parsed
+   * @return {@code true} if the current tool version is equal or less than the target version, {@code false} if
+   * it's larger
+   */
+  private boolean isEqualOrLess(int[] currentToolVersion, int[] targetVersion) {
+
+    for (int i = 0; i < Math.min(currentToolVersion.length, targetVersion.length); i++) {
+      if (currentToolVersion[i] < targetVersion[i]) {
+        return true;
+      } else if (currentToolVersion[i] > targetVersion[i]) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
Couldn't this be done with `VersionObject` e.g. `compareTo` method?

Reply:
Also fixed by using the VersionIdentifier and deleted this method completely.